### ISSUE TITLE
Costa requirement missing

### ISSUE THREAD

**Issue by matt-chan** [electronic-structure/SIRIUS#781]
Hi,

The build scripts and instructions should be updated to mention the requirement of COSTA. Currently the build fails because CMake can't find it. 

https://github.com/electronic-structure/SIRIUS/blob/master/cmake/siriusConfig.cmake.in#L41

Thanks!
Matt

**Comment by ltalirz** [comment]
To add some more details here: we ran into this build error when building q-e-sirius with nlcglib.

The problem is actually not in the spack recipe in the sirius repository, but that the sirius spack package in the nlcglib https://github.com/simonpintarelli/nlcglib/tree/develop/spack/packages is out of date.


@simonpintarelli could you please update it? Feel free to transfer this issue to the nlcglib repository.

This highlights another minor issue, namely that the [default nlcglib installation instructions](https://github.com/simonpintarelli/nlcglib#installation) point to `q-e-sirious@develop-ristretto`, which in turn points to `sirius@develop`, which is a moving target.

Perhaps the default installation instructions should install released versions of sirius?


the spack package points at the `develop` branch of sirius, which is a moving target.
Perhaps the spack package in the nlcglib repository should point to a fixed version of sirius (if you do want to vendor it).




 and we run into the build error.

**Comment by simonpintarelli** [comment]
@ltalirz @matt-chan I've update the recipes. Please use the develop version for the moment, I need to merge it with the latest wave function refactoring first. I'm closing it, please reopen in https://github.com/simonpintarelli/nlcglib if you hit another issue.

**Comment by ltalirz** [comment]
Thanks a lot for the quick fix, Simon!

**Comment by toxa81** [comment]
This is still present on the conda-forge
```
-- Found SCALAPACK: $PREFIX/lib/libscalapack.so  
CMake Error at CMakeLists.txt:197 (find_package):
  Could not find a package configuration file provided by "costa" with any of
  the following names:

    costaConfig.cmake
    costa-config.cmake

  Add the installation prefix of "costa" to CMAKE_PREFIX_PATH or set
  "costa_DIR" to a directory containing one of the above files.  If "costa"
  provides a separate development package or SDK, be sure it has been
  installed.

```
does the base image contain COSTA library?

**Comment by ltalirz** [comment]
Indeed, it appears COSTA was introduced as a default dependency of sirius in a patch release (from 7.3.1 to 7.3.2). I've added it to the dependencies now, let's see ... https://github.com/conda-forge/sirius-feedstock/pull/16

**Comment by ltalirz** [comment]
Hi,

just wanted to report that, for me, the sirius recipe from the nlcglib repo works in order to build q-e-sirius with nlcglib, but the vanilla sirius recipe from spack does not.

There are currently three different spack recipes for sirius:
* here in the repo: https://github.com/electronic-structure/SIRIUS/blob/master/spack/packages/sirius/package.py
* in nlcglib repo https://github.com/simonpintarelli/nlcglib/blob/develop/spack/packages/sirius/package.py
* spack upstream https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/sirius/package.py

all of which are slightly different, and users may struggle to figure out which one to use.

It would be great if those could be consolidated and upstreamed to spack

**Comment by simonpintarelli** [comment]
Hi Leo, I agree, for end-users the only relevant recipe should be the one in official spack. I'm currently rebasing my latest changes and will update the main spack recipe once this is done. The spack recipe in the SIRIUS repo is primarily for the CI.

**Comment by toxa81** [comment]
Sorry guys if I missed the discussion. Is the issue resolved?

**Comment by ltalirz** [comment]
@conradjohnston any issues with building with the spack recipe?

**Comment by toxa81** [comment]
I will close this one as inactive.
